src/node_buffer.cc

```diff
@@ -994,10 +994,12 @@ void IndexOfString(const FunctionCallbackInfo<Value>& args) {
   bool is_forward = args[4]->IsTrue();
 
   const char* haystack = ts_obj_data;
-  const size_t haystack_length = ts_obj_length;
-  // Extended latin-1 characters are 2 bytes in Utf8.
+  // Round down to the nearest multiple of 2 in case of UCS2.
+  const size_t haystack_length = (enc == UCS2) ?
+    ts_obj_length & ~1 : ts_obj_length;
```

Comment:
@trevnorris Because it’s a bit operation and using two’s complement seems unnecessary for that. I’d have written `&~ 1` if the linter had let me. Using `-2` is a nice trick, but I don’t consider it very idiomatic… If you prefer, I can use something like `AlignDown(ts_obj_length, sizeof(uint16_t))`.